### PR TITLE
feat: collections, auth HOCs

### PR DESCRIPTION
**Pull request by ndt080**

### CODE REVIEW

File: src/features/auth/components/WithAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+function WithAuth({ children }: PropsWithChildren) {
+  const { authorized } = useAuth();
+  const [isVisible, setIsVisible] = useState(true);
+
+  useEffect(() => {
+    setIsVisible(authorized);
+  }, [authorized]);
+
+  return <>{isVisible ? children : null}</>;
```
Comment (kirillkurko) [inline src/features/auth/components/WithAuth.tsx]: With the current `isVisible` name, you need to search for how state is changed to actually understand the logic here (it's pretty simple, but anyway it requires some cognitive load to process it). Maybe it makes sense to rename the state to `isAuthorized`, so there will be a single state, so you don't need to think how `isVisible` and `authorized` are connected

File: src/features/auth/components/WithoutAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+function WithoutAuth({ children }: PropsWithChildren) {
+  const { authorized } = useAuth();
+  const [isVisible, setIsVisible] = useState(true);
+
+  useEffect(() => {
+    setIsVisible(!authorized);
+  }, [authorized]);
+
+  return <>{isVisible ? children : null}</>;
```
Comment (kirillkurko) [inline src/features/auth/components/WithoutAuth.tsx]: Please check the that I left for `WithAuth` component, I think that we can slightly improve readability here

File: src/features/collections/components/CollectionList/CollectionList.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }, [type]);
+
+  const resetCollection = async () => {
+    setPage(1);
+    setItems([]);
+    await fetchMoreItems();
+  };
+
+  const fetchMoreItems = async () => {
+    console.log('load');
```
Comment (kirillkurko) [inline src/features/collections/components/CollectionList/CollectionList.tsx]: Do we need to keep this `console.log`?

File: src/features/collections/values/collection-tabs.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  [CollectionType.WatchLater]: {
+    id: 1,
+    title: 'Watchlist',
+  },
+  [CollectionType.Watched]: {
+    id: 2,
+    title: 'Completed',
+  },
+} as Record<CollectionType, { id: number; title: string }>;
```
Comment (kirillkurko) [inline src/features/collections/values/collection-tabs.ts]: Cannot we use `satisfies` here or change it like in the snippet below to achieve better type-safety here?
```typescript
export const CollectionTabs: Record<CollectionType, { id: number; title: string }> = {
// ...
}
```

**Review by kirillkurko — APPROVED** [review]
Looks good to me, I left just a couple of small non-blocking comments, feel free to merge this one once you address them 🚀

File: src/features/collections/components/CollectionList/CollectionList.tsx (hunk 3 above)
Comment (ndt080) [reply src/features/collections/components/CollectionList/CollectionList.tsx]: not needed)

File: src/features/auth/components/WithAuth.tsx (hunk 1 above)
Comment (ndt080) [reply src/features/auth/components/WithAuth.tsx]: yes, isAuthorized would be a more best name for the state

File: src/features/collections/values/collection-tabs.ts (hunk 4 above)
Comment (ndt080) [reply src/features/collections/values/collection-tabs.ts]: @kirillkurko Сorrected the typing

```
export type CollectionTabsItem = { id: number; title: string };

export const CollectionTabs: Record<CollectionType, CollectionTabsItem | null> = {
  [CollectionType.Favorite]: {
    id: 0,
    title: 'Favorite',
  },
  ...
  [CollectionType.History]: null,
};
```

**Pull request merged**